resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/backend/minikube/Minikube.scala

```diff
@@ -56,60 +64,106 @@ private[spark] object Minikube extends Logging {
         .getOrElse(throw new IllegalStateException(s"Unknown status $statusString"))
     }
   }
 
   def getKubernetesClient: DefaultKubernetesClient = {
+    // only the three-part version number is matched (the optional suffix like "-beta.0" is dropped)
+    val versionArrayOpt = "\\d+\\.\\d+\\.\\d+".r
+      .findFirstIn(minikubeVersionString.split(VERSION_PREFIX)(1))
+      .map(_.split('.').map(_.toInt))
+
+    assert(versionArrayOpt.isDefined && versionArrayOpt.get.size == 3,
+      s"Unexpected version format detected in `$minikubeVersionString`." +
+      "For minikube version a three-part version number is expected (the optional non-numeric " +
+      "suffix is intentionally dropped)")
+
+    val kubernetesConf = versionArrayOpt.get match {
+      case Array(x, y, z) =>
+        // comparing the versions as the kubectl command is only introduced in version v1.1.0:
+        // https://github.com/kubernetes/minikube/blob/v1.1.0/CHANGELOG.md
+        if (Ordering.Tuple3[Int, Int, Int].gteq((x, y, z), (1, 1, 0))) {
+          kubectlBasedKubernetesClientConf
+        } else {
+          legacyKubernetesClientConf
+        }
+    }
+    new DefaultKubernetesClient(kubernetesConf)
+  }
+
+  private def legacyKubernetesClientConf: Config = {
     val kubernetesMaster = s"https://${getMinikubeIp}:8443"
     val userHome = System.getProperty("user.home")
-    val minikubeBasePath = Paths.get(userHome, MINIKUBE_PATH).toString
-    val profileDir = if (Files.exists(Paths.get(minikubeBasePath, "apiserver.crt"))) {
-      // For Minikube <1.9
-      ""
-    } else {
-      // For Minikube >=1.9
-      Paths.get("profiles", executeMinikube("profile")(0)).toString
-    }
-    val apiServerCertPath = Paths.get(minikubeBasePath, profileDir, "apiserver.crt")
-    val apiServerKeyPath = Paths.get(minikubeBasePath, profileDir, "apiserver.key")
-    val kubernetesConf = new ConfigBuilder()
-      .withApiVersion("v1")
-      .withMasterUrl(kubernetesMaster)
-      .withCaCertFile(
-        Paths.get(userHome, MINIKUBE_PATH, "ca.crt").toFile.getAbsolutePath)
-      .withClientCertFile(apiServerCertPath.toFile.getAbsolutePath)
-      .withClientKeyFile(apiServerKeyPath.toFile.getAbsolutePath)
+    buildKubernetesClientConf(
+      "v1",
+      kubernetesMaster,
+      Paths.get(userHome, MINIKUBE_PATH, "ca.crt").toFile.getAbsolutePath,
+      Paths.get(userHome, MINIKUBE_PATH, "apiserver.crt").toFile.getAbsolutePath,
+      Paths.get(userHome, MINIKUBE_PATH, "apiserver.key").toFile.getAbsolutePath)
+  }
+
+  private def kubectlBasedKubernetesClientConf: Config = {
+    val outputs = executeMinikube("kubectl config view")
+    val apiVersionString = outputs.find(_.contains(APIVERSION_PREFIX))
+    val serverString = outputs.find(_.contains(SERVER_PREFIX))
+    val caString = outputs.find(_.contains(CA_PREFIX))
+    val clientCertString = outputs.find(_.contains(CLIENTCERT_PREFIX))
+    val clientKeyString = outputs.find(_.contains(CLIENTKEY_PREFIX))
```

Comment:
Well, this logic seems to assume `kubectl config view` has only a single config. We should assume like that.
Technically, there are multiple configurations and `minikube` is just specified as the current context via `current-context: minikube`.